### ISSUE TITLE
[FEATURE] how to return mlx intermediate layer output similarly to Keras 

### ISSUE THREAD

**Issue by thegodone** [ml-explore/mlx#1056]
**Describe the FEATURE**

Is there a way to get the intermediate tensor results similar to keras following code. I try to determine why the mlx/ Keras code are not returning the same output. This is almost one week that I work on it and when I apply keras pretrained values I got a completely different result. I would like to determine where the drift occur and fix it without making N intermediates models and reuse the final Keras model weights.

**To Reproduce**

Include code snippet
```

from tensorflow.keras.models import Model

def get_intermediate_outputs_keras(model, input_data):
    layer_outputs = [layer.output for layer in model.layers]  # all layer outputs
    intermediate_model = Model(inputs=model.input, outputs=layer_outputs)
    intermediate_predictions = intermediate_model.predict(input_data)
    return intermediate_predictions

```

**Expected behavior**

**Desktop (please complete the following information):**
 - OS Version: MacOS 14.4
 - Version 0.7.0
  

**Additional context**

**Comment by thegodone** [comment]
Yes your proposal is interesting. That would be nice to have access to the
layers in order to run this DebugWrapper class. In keras this is of course
easy.

**Comment by thegodone** [comment]
any update on that implementation ? cause so far DebugWrapper is not working ;-) or I miss something ?

**Comment by angeloskath** [comment]
Can you let us know what is not working with the above example?

**Comment by thegodone** [comment]
DebugWrapper is not implemented.

**Comment by zcbenz** [comment]
I'm closing this issue since solutions had been provided and there has been no similar requests that would make us consider providing specialized APIs.
